vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailFragment.kt

```diff
@@ -793,6 +794,24 @@ class RoomDetailFragment @Inject constructor(
                 .show()
     }
 
+    private fun promptReasonToDeleteEvent(eventId: String) {
+        val layout = requireActivity().layoutInflater.inflate(R.layout.dialog_delete_event, null)
+        val reasonCheckBox = layout.findViewById<MaterialCheckBox>(R.id.deleteEventReasonCheck)
+        val reasonInput = layout.findViewById<TextInputEditText>(R.id.deleteEventReasonInput)
+
+        reasonCheckBox.setOnCheckedChangeListener { _, isChecked -> reasonInput.isEnabled = isChecked }
+
+        AlertDialog.Builder(requireActivity())
+                .setView(layout)
+                .setTitle(R.string.delete_event_dialog_title)
+                .setPositiveButton(R.string.remove) { _, _ ->
+                    val reason = if (reasonCheckBox.isChecked) reasonInput.text.toString().takeIf { it.isNotBlank() } else null
```

Comment:
Would be nicer to use two `takeIf()` statements:
```
reasonInput.text.toString()
    .takeIf { reasonCheckBox.isChecked }
    .takeIf { it.isNotBlank() }
```

Reply:
This is brilliant suggestion.